fix: Apply organization brand colors and theme to member personal events

anikdhabal opened the pull request:
## What does this PR do?

Fixes organization brand colors and theme not applying to personal events of organization members. Previously, org-level brand colors and themes only applied to team events but were ignored for personal events, causing inconsistent branding within organizations.

devin-ai-integration[bot] committed:
fix: Apply organization brand colors and theme to member personal events

- Fetch organization brand colors and theme in getEventTypesFromDB
- Override user brand colors/theme with org values for personal events
- Apply to booking confirmation pages, user booking pages, and routing forms
- Follow same pattern as hideBranding for consistency
- Fixes issue where org brand colors only applied to team events

Co-Authored-By: anik@cal.com <adhabal2002@gmail.com>

devin-ai-integration[bot] committed:
fix: Resolve type errors for organization brand color and theme overrides

- Added brandColor, darkBrandColor, theme to ProfileRepository findById organization select
- Updated test mock to include new organization fields
- Ensures consistent organization branding types across all queries
- All type errors related to brand color changes now resolved

Co-Authored-By: anik@cal.com <adhabal2002@gmail.com>

devin-ai-integration[bot] committed:
fix: Remove redundant profile assignment causing type errors

The spread operator already includes profile when it exists.
Explicit assignment causes type conflicts with union types where
profile is not guaranteed to exist in all branches.

Resolves TypeScript error at bookings-single-view.getServerSideProps.tsx:134

Co-Authored-By: anik@cal.com <adhabal2002@gmail.com>

devin-ai-integration[bot] committed:
fix: Use eventTypeRaw.profile for organization brand color access

Changed eventType.profile to eventTypeRaw.profile in profile object construction
to fix TypeScript errors where profile property doesn't exist on transformed
eventType object. The eventTypeRaw object maintains the original profile field
from the database query.

Resolves TypeScript errors at lines 155, 158, 161 in
bookings-single-view.getServerSideProps.tsx

Co-Authored-By: anik@cal.com <adhabal2002@gmail.com>

anikdhabal committed:
fix type error

hariombalhara commented on apps/web/modules/bookings/views/bookings-single-view.getServerSideProps.tsx:
```diff
@@ -150,9 +150,15 @@ export async function getServerSideProps(context: GetServerSidePropsContext) {
+      : eventType.profile?.organization?.brandColor ?? eventType.users[0].brandColor ?? null,
+    darkBrandColor: eventType.team
+      ? null
+      : eventType.profile?.organization?.darkBrandColor ?? eventType.users[0].darkBrandColor ?? null,
```
I think it would be best to abstract the theming logic(including branding colors) out into a  utility like _getBranding_ or similar.
then we could do something like following everywhere.

```
...getBranding({eventType})
```

hariombalhara reviewed (changes requested):
Left a suggestion

devin-ai-integration[bot] committed:
refactor: Extract branding logic into getBrandingForEventType utility

- Create getBrandingForEventType utility function that handles organization
  brand color overrides for both team and personal events
- For team events: org branding → team branding → null
- For personal events: org branding → user branding → null
- Refactor bookings-single-view to use utility function
- Update team booking page to apply organization brand overrides
- Add organization branding fields to ProfileRepository and team queries

Addresses review feedback from @hariombalhara on PR #24456

Co-Authored-By: anik@cal.com <adhabal2002@gmail.com>

anikdhabal committed:
Update getBranding.ts

anikdhabal committed:
Change

Set themeBasis to null instead of branding.theme.

hariombalhara commented on apps/web/lib/team/[slug]/[type]/getServerSideProps.tsx:
```diff
@@ -133,6 +134,23 @@ export const getServerSideProps = async (context: GetServerSidePropsContext) =>
+              theme: team.parent.theme,
+            },
+          }
+        : team,
```
nit: I would have prefered if it could be simplified to something like. 

e.g. youy don't need to pick brandColor for team then why is it needed for team.parent
``team: team.parent ?? team``

hariombalhara commented on packages/features/profile/lib/getBranding.ts:
```diff
@@ -0,0 +1,48 @@
+      theme: eventType.team.parent?.theme ?? eventType.team.theme ?? null,
+      brandColor: eventType.team.parent?.brandColor ?? eventType.team.brandColor ?? null,
+      darkBrandColor: eventType.team.parent?.darkBrandColor ?? eventType.team.darkBrandColor ?? null,
+    };
```
I am not sure if this fallback is entirely correct.

My confusion:
What if _eventType.team.parent.theme_ is set but _eventType.team.parent?.brandColor_ isn't. Then we endup picking one thing from organization and other thing from team.


I think the right approach would be to check if parent exists and if it does use all the settings of it or use none.

```
const branding = team.parent ?? team

theme: branding.theme,
brandColor: branding.brandColor
darkBrandColor: branding.darkBrandColor
```

Similar is true for user branding

hariombalhara commented on apps/web/server/lib/[user]/getServerSideProps.ts:
```diff
@@ -135,10 +138,11 @@ export const getServerSideProps: GetServerSideProps<UserPageProps> = async (cont
     avatarUrl: user.avatarUrl,
-    darkBrandColor: user.darkBrandColor ?? DEFAULT_DARK_BRAND_COLOR,
+    darkBrandColor:
+      user.profile.organization?.darkBrandColor ?? user.darkBrandColor ?? DEFAULT_DARK_BRAND_COLOR,
```
Can we not use the similar abstraction here. Like we could have one to get it for a user within the same modiule(getBranding)

devin-ai-integration[bot] committed:
refactor: Simplify branding fallback logic to prevent mixing org/team settings

- Changed getBrandingForEventType to use either parent or team branding entirely
- Simplified team parent branding structure in getServerSideProps
- Prevents inconsistent branding when some properties exist in parent but not others
- Addresses PR review feedback from hariombalhara

Co-Authored-By: hariom@cal.com <hariombalhara@gmail.com>

anikdhabal committed:
addressed review

devin-ai-integration[bot] committed:
test: Add comprehensive unit tests for branding utility functions

- Add 20 unit tests covering getBrandingForEventType, getBrandingForUser, and getBrandingForTeam
- Test organization branding override scenarios for both team and personal events
- Test fallback behavior when organization branding is not set
- Test handling of null and optional branding properties
- Verify theme and color inheritance from parent organizations

Co-Authored-By: anik@cal.com <adhabal2002@gmail.com>

devin-ai-integration[bot] committed:
test: Simplify branding tests to focus on core scenarios

- Reduce from 20 to 8 essential test cases
- Focus on org override and fallback behavior for each utility
- Remove redundant edge case tests while maintaining full coverage

Co-Authored-By: anik@cal.com <adhabal2002@gmail.com>

anikdhabal committed:
Merge branch 'main' into devin/fix-org-brand-colors-1760372393
